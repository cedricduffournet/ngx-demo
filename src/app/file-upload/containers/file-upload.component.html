<div class="message">
  <input #file type="file" (change)="uploadFile($event)" accept="image/png, image/jpeg">
</div>