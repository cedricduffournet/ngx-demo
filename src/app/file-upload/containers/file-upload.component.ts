import { Component, OnInit } from '@angular/core';
import { select, Store } from '@ngrx/store';
import { Observable } from 'rxjs';

import * as fromFileUpload from '@app/file-upload/state/reducers';
import { FileUploadActions } from '@app/file-upload/state/actions';
// import * as fromFileUploadSelectors from 'src/app/upload-file-store/selectors';
// import * as fromFileUploadState from 'src/app/upload-file-store/state';

@Component({
  selector: 'app-file-upload',
  templateUrl: './file-upload.component.html'/* ,
  styleUrls: ['./file-upload.component.css'] */
})
export class FileUploadComponent implements OnInit {
  completed$: Observable<boolean>;
  progress$: Observable<number | null>;
  error$: Observable<string | null>;
  isInProgress$: Observable<boolean>;
  isReady$: Observable<boolean>;
  hasFailed$: Observable<boolean>;

  constructor(private store: Store<fromFileUpload.State>) {}

  ngOnInit() {
    /*     this.completed$ = this.store.pipe(
      select(fromFileUploadSelectors.selectUploadFileCompleted)
    ); */

    this.progress$ = this.store.pipe(
      select(fromFileUpload.getFileUploadProgress)
    );

    this.error$ = this.store.pipe(select(fromFileUpload.getFileUploadError));

    /*     this.isInProgress$ = this.store$.pipe(
      select(fromFileUploadSelectors.selectUploadFileInProgress)
    );

    this.isReady$ = this.store$.pipe(
      select(fromFileUploadSelectors.selectUploadFileReady)
    );

    this.hasFailed$ = this.store$.pipe(
      select(fromFileUploadSelectors.selectUploadFileFailed)
    ); */
  }

  uploadFile(event: any) {
    const files: FileList = event.target.files;
    const file = files.item(0) as File;
    console.log(file);
    const sendFile = {
      ...file,
      name: file.name
    };

    console.log(sendFile);
    /*     const files: FileList = event.target.files;
    
 */


    this.store.dispatch(FileUploadActions.uploadRequest({file}));
/*       new fromFileUploadActions.UploadRequestAction({
        file
      })
    ); */

    // clear the input form
    event.srcElement.value = null;
  }

  resetUpload() {
    this.store.dispatch(FileUploadActions.uploadReset() );
    // this.store$.dispatch(new fromFileUploadActions.UploadResetAction());
  }

  cancelUpload() {
    this.store.dispatch(FileUploadActions.uploadCancel() );
    // this.store$.dispatch(new fromFileUploadActions.UploadCancelAction());
  }
}
