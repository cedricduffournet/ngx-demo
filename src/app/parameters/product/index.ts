export * from './parameter-product.component';
