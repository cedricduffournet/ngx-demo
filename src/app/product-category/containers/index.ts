export * from './product-category-add-modal/product-category-add-modal.component';
export * from './product-category-update-modal/product-category-update-modal.component';
export * from './product-category-delete-modal/product-category-delete-modal.component';
export * from './product-category-list-view/product-category-list-view.component';
