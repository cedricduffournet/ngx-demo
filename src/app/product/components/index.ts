export * from './product-add/product-add.component';
export * from './product-update/product-update.component';
export * from './product-delete/product-delete.component';
export * from './product-form/product-form.component';
export * from './product-item/product-item.component';
export * from './product-items/product-items.component';
export * from './product-details/product-details.component';
