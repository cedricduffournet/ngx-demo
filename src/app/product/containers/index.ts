export * from './product-add-view/product-add-view.component';
export * from './product-update-modal/product-update-modal.component';
export * from './product-delete-modal/product-delete-modal.component';
export * from './product-list-view/product-list-view.component';
export * from './product-selected/product-selected.component';
export * from './product-details-view/product-details-view.component';
