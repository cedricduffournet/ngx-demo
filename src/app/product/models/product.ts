import { schema } from 'normalizr';
import { ProductCategory, productCategorySchema } from '@app/product-category/models/product-category';

export interface Product {
  id: number;
  name: string;
  categories: ProductCategory[];
}

export const productSchema = new schema.Entity('products', {
  categories: [productCategorySchema]
});
