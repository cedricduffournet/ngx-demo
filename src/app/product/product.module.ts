import { NgModule, ComponentFactoryResolver } from '@angular/core';

import { RouterModule } from '@angular/router';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';

import { SharedModule } from '@app/shared/shared.module';
import { PipesModule } from '@app/shared/pipes/pipe.module';

import { ProductCategoryModule } from '@app/product-category/product-category.module';
import { ProductEffects } from '@app/product/state/effects';
import { ProductFacade } from '@app/product/state/product.facade';
import {
  ProductAddComponent,
  ProductUpdateComponent,
  ProductDeleteComponent,
  ProductFormComponent,
  ProductItemComponent,
  ProductItemsComponent,
  ProductDetailsComponent
} from '@app/product/components';
import {
  ProductAddViewComponent,
  ProductUpdateModalComponent,
  ProductDeleteModalComponent,
  ProductListViewComponent,
  ProductSelectedComponent,
  ProductDetailsViewComponent
} from '@app/product/containers';
import { reducers } from '@app/product/state/reducers';
import { ProductGuard } from '@app/product/services/product.guard';
import { CoalescingComponentFactoryResolver } from '@app/coalescing-component-factory-resolver.service';

@NgModule({
  imports: [
    RouterModule,
    SharedModule,
    PipesModule,
    ProductCategoryModule,
    EffectsModule.forFeature([ProductEffects]),
    StoreModule.forFeature('products', reducers)
  ],
  declarations: [
    ProductListViewComponent,
    ProductAddComponent,
    ProductUpdateComponent,
    ProductDeleteComponent,
    ProductFormComponent,
    ProductItemComponent,
    ProductItemsComponent,
    ProductAddViewComponent,
    ProductUpdateModalComponent,
    ProductDeleteModalComponent,
    ProductDetailsComponent,
    ProductSelectedComponent,
    ProductDetailsViewComponent
  ],
  entryComponents: [
    ProductUpdateModalComponent,
    ProductDeleteModalComponent
  ],
  exports: [ProductListViewComponent, ProductAddViewComponent, ProductDetailsViewComponent],
  providers: [ProductFacade, ProductGuard]
})
export class ProductModule {
  // see https://github.com/angular/angular/issues/14324
  constructor(
    coalescingResolver: CoalescingComponentFactoryResolver,
    localResolver: ComponentFactoryResolver
  ) {
    coalescingResolver.registerResolver(localResolver);
  }
}
