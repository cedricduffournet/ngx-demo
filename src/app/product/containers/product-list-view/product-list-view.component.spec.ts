import { TestBed, ComponentFixture } from '@angular/core/testing';

import { provideMockStore } from '@ngrx/store/testing';
import { TranslateModule } from '@ngx-translate/core';

import {
  ProductItemComponent,
  ProductItemsComponent
} from '@app/product/components';
import { ProductListViewComponent } from '@app/product/containers';
import { SharedModule } from '@app/shared/shared.module';
import { ProductFacade } from '@app/product/state/product.facade';
import { AuthFacade } from '@app/authentication/state/auth.facade';

describe('ProductListViewComponent', () => {
  let fixture: ComponentFixture<ProductListViewComponent>;
  let component: ProductListViewComponent;
  let facade: ProductFacade;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [
        ProductListViewComponent,
        ProductItemsComponent,
        ProductItemComponent
      ],
      imports: [SharedModule, TranslateModule.forRoot()],
      providers: [
        provideMockStore(),
        ProductFacade,
        AuthFacade
      ]
    });

    fixture = TestBed.createComponent(ProductListViewComponent);
    component = fixture.componentInstance;
    facade = TestBed.get(ProductFacade);
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should loadProduct on init', () => {
    spyOn(facade, 'loadProducts');
    fixture.detectChanges();
    expect(facade.loadProducts).toHaveBeenCalledWith();
  });

  it('should call showAddProductModal on add event', () => {
    spyOn(facade, 'showAddProductModal');
    component.onAdd();
    expect(facade.showAddProductModal).toHaveBeenCalledWith();
  });

  it('should call showUpdateProductModal on update event', () => {
    spyOn(facade, 'showUpdateProductModal');
    component.onUpdate(1);
    expect(facade.showUpdateProductModal).toHaveBeenCalledWith();
  });

  it('should call selectProduct on update event', () => {
    spyOn(facade, 'selectProduct');
    component.onUpdate(1);
    expect(facade.selectProduct).toHaveBeenCalledWith(1);
  });

  it('should call showDeleteProductModal on update event', () => {
    spyOn(facade, 'showDeleteProductModal');
    component.onDelete(1);
    expect(facade.showDeleteProductModal).toHaveBeenCalledWith();
  });

  it('should call selectProduct on delete event', () => {
    spyOn(facade, 'selectProduct');
    component.onDelete(1);
    expect(facade.selectProduct).toHaveBeenCalledWith(1);
  });

  it('should call changePage when page change', () => {
    spyOn(facade, 'changePage');
    component.onPageChanged(1);
    expect(facade.changePage).toHaveBeenCalledWith(1);
  });
});
