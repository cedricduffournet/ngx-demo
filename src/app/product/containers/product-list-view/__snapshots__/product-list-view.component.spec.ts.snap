// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListViewComponent should be created 1`] = `
<app-product-list-view
  authFacade={[Function AuthFacade]}
  canCreate$={[Function Store]}
  canDelete$={[Function Store]}
  canUpdate$={[Function Store]}
  config$={[Function Store]}
  facade={[Function ProductFacade]}
  products$={[Function Store]}
  totalItems$={[Function Store]}
>
  <app-page-inner>
    <div
      class="px-3 py-3 "
    >
      
      <div
        class="page-section"
      >
        <app-product-items
          ng-reflect-authorization="[object Object]"
        >
          <section
            class="bg-white border-left border-right border-bottom rounded"
          >
            <div
              class="table-responsive"
            >
              <table
                class="table table-hover"
              >
                <thead>
                  <tr>
                    <th
                      class="align-middle"
                      ng-reflect-translate=""
                      translate=""
                    >
                       NAME 
                    </th>
                    <th
                      class="align-middle text-right"
                    >
                       Actions 
                    </th>
                  </tr>
                </thead>
                <tbody>
                  
                </tbody>
              </table>
            </div>
          </section>
          
        </app-product-items>
        
      </div>
    </div>
  </app-page-inner>
</app-product-list-view>
`;
